sdk/python/feast/infra/online_stores/sqlite.py

```diff
@@ -127,31 +129,38 @@ def online_write_batch(
                 if progress:
                     progress(1)
 
+    @log_exceptions_and_usage(online_store="sqlite")
     def online_read(
         self,
         config: RepoConfig,
         table: Union[FeatureTable, FeatureView],
         entity_keys: List[EntityKeyProto],
         requested_features: Optional[List[str]] = None,
     ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
-        pass
         conn = self._get_conn(config)
         cur = conn.cursor()
 
         result: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
 
-        project = config.project
-        for entity_key in entity_keys:
-            entity_key_bin = serialize_entity_key(entity_key)
-
+        with tracing_span(name="remote_call"):
+            # Fetch all entities in one go
             cur.execute(
-                f"SELECT feature_name, value, event_ts FROM {_table_id(project, table)} WHERE entity_key = ?",
-                (entity_key_bin,),
+                f"SELECT entity_key, feature_name, value, event_ts "
+                f"FROM {_table_id(config.project, table)} "
+                f"WHERE entity_key IN ({','.join('?' * len(entity_keys))}) "
+                f"ORDER BY entity_key",
+                [serialize_entity_key(entity_key) for entity_key in entity_keys],
```

Comment:
This is done to avoid using `tracing_span` context manager in the loop. So I refactored it to have one call to sqlite instead. As a side effect this sped up sqlite retrieval by 10-15%. And this change seemed pretty small to me to create separate PR.

Reply:
Any concerns with query itself?

Reply:
Gotcha, no concerns with the query